python/lsst/daf/butler/queries/_base.py

```diff
@@ -242,7 +246,7 @@ def where(
 
     @abstractmethod
     def _get_datasets(self) -> Set[str]:
-        """Return all dataset types included in the query's result rows."""
+        """Return all datasets included in the query's result rows."""
```

Comment:
```suggestion
        """Return all dataset type names included in the query's result rows."""
```
Method name notwithstanding, these are more dataset types (they're the names of them) than datasets.